ZenPacks/zenoss/Microsoft/Windows/txwinrm_utils.py

```diff
@@ -56,96 +59,102 @@ def createConnectionInfo(device_proxy):
+    def getProxyValue(props):
+        if not isinstance(props, list):
+            props = [props]
+        for p in props:
+            if hasattr(device_proxy, p):
+                attr = getattr(device_proxy, p)
+                val = attr() if callable(attr) else attr
+                if val:
+                    return val
+        return ""
+
+    hostname = getProxyValue(['windows_servername', 'zWinRMServerName', 'manageIp'])
+    if not hostname:
+        raise UnauthorizedError("Attempted Windows connection to non-Windows device")
+
+    hostname = getProxyValue(['windows_servername', 'zWinRMServerName', 'manageIp'])
+
+    username = getProxyValue(['windows_user', 'zWinRMUser', 'zWinUser'])
     if not username:
-        raise UnauthorizedError("zWinRMUser or zWinUser must be configured")
+        raise UnauthorizedError(
+            "Cannot connect to Windows with an unknown user, zWinRMUser or zWinUser must be configured")
 
     # Warn about old-style usernames of the DOMAIN\User format.
-    pattern = r'[a-zA-Z0-9][a-zA-Z0-9.]{0,14}\\[^"/\\\[\]:;|=,+*?<>]{1,104}'
-    if re.match(pattern, username):
-        raise UnauthorizedError(
-            "zWinRMUser must be user@example.com, not DOMAIN\User")
+    if re.match(r'[a-zA-Z0-9][a-zA-Z0-9.]{0,14}\\[^"/\\\[\]:;|=,+*?<>]{1,104}', username):
+        raise UnauthorizedError("zWinRMUser must be user@example.com, not DOMAIN\User")
 
-    password = device_proxy.windows_password
+    password = getProxyValue(['windows_password', 'zWinRMPassword'])
     if not password:
-        raise UnauthorizedError(
-            "zWinRMPassword or zWinPassword must be configured")
+        raise UnauthorizedError("zWinRMPassword or zWinPassword must be configured")
 
+    winKDC = getProxyValue(['zWinKDC', 'zWinTrustedKDC'])
     auth_type = 'kerberos' if '@' in username else 'basic'
-    if auth_type == 'kerberos' and not device_proxy.zWinKDC:
-        raise UnauthorizedError(
-            "zWinKDC must be configured for domain authentication")
+    if auth_type == 'kerberos' and not winKDC:
+        raise UnauthorizedError("zWinKDC must be configured for domain authentication")
 
-    scheme = device_proxy.zWinScheme.lower()
+    scheme = getProxyValue('zWinScheme')
+    scheme = scheme if hasattr(scheme, 'lower') else ''
     if scheme not in ('http', 'https'):
-        raise UnauthorizedError(
-            "zWinScheme must be either 'http' or 'https'")
-
-    ok_ports = (5986, 443)
-    if int(device_proxy.zWinRMPort) not in ok_ports and scheme == 'https':
+        raise UnauthorizedError("zWinScheme must be either 'http' or 'https'")
+
+    try:
+        port = int(getProxyValue('zWinRMPort'))
+        if port not in (5986, 443) and scheme == 'https':
+            raise Exception()
+        if port not in (5985, 80) and scheme == 'http':
+            raise Exception()
+    except Exception:
         raise UnauthorizedError("zWinRMPort must be 5986 or 443 if zWinScheme is https")
```

Comment:
Line 106 checks for http and port (80, 443) but error message does not state http configuration is a valid setting.

Reply:
correct.  I extended the error message